### DISCUSSION TITLE
Routing User Requests in Qiskit Slack to IBMers

### DISCUSSION THREAD

**Discussion by jwoehr** [Qiskit/.github#16] [Ideas]
When users ask questions in Qiskit Slack that require IBM response or intervention, how do Advocates and other helpful folks route these requests?  I [commented](https://qiskit.slack.com/archives/GMT338B52/p1641825549002600) as follows in Qiskit Slack qiskit-advocates channel:

> WIBNI we had a spreadsheet of who to ask about what components of IBM Quantum so when a user asks a question that requires an IBMer to answer we knew whom to ping?

These questions include things like account blockage questions, components of IBM Q Experience that are down, sophisticated programming questions that probably should receive some informal q-and-a before popping up in the always-overloaded Issue tracker, etc.

Handling these user requests is a non-linear problem with several interacting dynamics:

- It's good for users to know that their requests are heard and that an Advocate or Community Organizer is routing their request to the appropriate developer of administrator.
- IBM developers and administrators must not become overloaded by trivial requests.
- In theory, the community organizers should be able to route requests, but they cannot be everywhere, and they do not provide completely overlapping coverage. Also, IBM USA-oriented holiday leaves can seem long, especially to our large contingent of advanced users in countries that observe very different holiday schedules. My own inclination has always been to @-ping some IBMer who of course won't answer until after the holidays, but the user sees that their question is getting routed and is satisfied for the moment.

Is there a more systematic way to route user requests to IBM attention than that _ad hoc_ way we do it now?

**Comment by javabster** [comment]
This is a great discussion topic @jwoehr! As you can tell from the slow response here the holiday scheduling part is definitely valid 😅 

Another dynamic is that IBM Quantum is a fast moving team and people do move around different specialisms from time to time, so finding the right person to talk to _that is still the right person to talk to_ can be challenging even internally.

I would be interested in hearing more from the community about what they would find the most useful mechanism for this kind of thing. Just getting a concrete list together fo what type of issues they most need help with would be a good first step. We do have [this page](https://github.com/qiskit-community/feedback/wiki/Communication-Channels-&-Support) in the wiki of this repo about where people can go for support, but perhaps there is more detail that could be added, or a different place to put this info where people can more easily find it

**Reply by jwoehr** [reply]
All the info about "where to get help" really comes down to "Read the documentation, and if you're confused, ask on Qiskit Slack."
Which people do: they ask on Slack. Some people get help. Some people get ignored. 
If I think a user's question is worth an answer from an IBMer, who's the "duty officer"?
Mostly at this point if I don't know who does what, I try to flag Junye.
IBMers seem to be participating less and less.
Which I understand because Qiskit Slack is overwhelming at this point, almost 25000 accounts in the `#general` channel.
"It costs a lot to win," sang the Grateful Dead!

**Reply by jwoehr** [reply]
@javabster [here's a link to one of those questions](https://qiskit.slack.com/archives/C7SJ0PJ5A/p1644291017994359) that it's very difficult for even a fairly well-informed user to answer for themselves, and for which a Google search will return any number of misleading out-of-date Qiskit documentation.

**Reply by jwoehr** [reply]
In particular, @javabster , the `#applications` channel doesn't seem to be getting adequately serviced. It's a class of questions I myself am not very good at answering, even partially.

**Reply by jwoehr** [reply]
The other constant stream of questions is the one about "What happened to _x_ on IBM Q Experience" which of course only IBMers can answer. Sometimes the answering lags a bit.

**Comment by jwoehr** [comment]
Glad we had a chance to talk about this today in [Qiskit Office Hours](https://github.com/qiskit-community/feedback/discussions/17)!
